equetzal/huronOS-build-tools
# Adding software-modules to sysforge

Adding software-modules to sysforge

## Why is this PR needed?
Currently huronOS requires from human interaction to build each one of the software-modules contained in the `software-modules` directory of this repostory.

Also, sysforge being *script-agnostic* is capable of doing this task automatically, so it's possible to improve the build times and the human interation needed to fully create all the software-modules required in the system.

## What have changed?
Now sysforge require flags:
- `sysforge --system-layers` for only the system layers build.
- `sysforge --software-modules` for only the software modules build
- Both flags to build all.

This have improved significantly the build times, by testing I archieved a 58 minutes build time since launching *sysforge* and building from layer 02-firmware all the way up to all the software modules.

This significantly improves the human times which used to tend into 2-3 hours and required constant interaction.

## How to test
Build the 01-core.hsl, install it on usb drive; Make sure to have automatic internet connectivity (e.g. ethernet with DHCP, USB tethering, etc).

 Then launch
```bash
sysforge --system-layers --software-modules
```
Fill the prompts and accept.

Wait for the build to finish (The wallpaper is only visible when the build have been finished).

Validate that all the modules have been created, the system layers are properly mounted, etc.
